### PR TITLE
GH-43570: [CI][Dev][Docs] Update references to "docker-compose" with "docker compose"

### PR DESCRIPTION
**Pull request by amoeba**
### Rationale for this change

This is a follow-on to https://github.com/apache/arrow/pull/43566 which was merged earlier today. See https://github.com/apache/arrow/issues/43570 for more detail on the entire set of changes.

### What changes are included in this PR?

- Replace instances where we call or document calling `docker-compose` with `docker compose`.

### Are these changes tested?

Yes.

### Are there any user-facing changes?

This PR includes the require user-facing changes to document things.
* GitHub Issue: #43570

### CODE REVIEW

**Comment by pitrou** [comment]
Note that more Archery changes were done in PR #43586.

**Comment by amoeba** [comment]
Yeah. Let me see what's needed to close this out now.

**Comment by amoeba** [comment]
I rebased and went through the repo again to find any instances that needed attention. Waiting for CI but I think this is ready for a review.

**Comment by amoeba** [comment]
Pushed one commit to fix the archery tests.

**Comment by amoeba** [comment]
The three CI failures don't look related to me.

File: docs/source/developers/continuous_integration/archery.rst
Comment (kou) [inline docs/source/developers/continuous_integration/archery.rst:39]:
```suggestion
and `Docker Compose <https://docs.docker.com/compose/>`_, which you may
```

**Review by kou — APPROVED** [review]
+1

File: docs/source/developers/continuous_integration/archery.rst
Comment (amoeba) [reply docs/source/developers/continuous_integration/archery.rst]: Thanks!

**Comment by amoeba** [comment]
@kou 's suggestion encouraged me to do a more serious pass, trying to use "Docker Compose" everywhere I saw we were referring to the product/software rather than a shell command. See https://github.com/apache/arrow/pull/43575/commits/71a365a52c5b921aa2828d440e3b4500b6a1d861. Some calls were subjective, though I think the most important thing has been done here (change `docker-compose` to `docker compose` for all example shell commands).

@pitrou do you want to take a last look here?

**Review by pitrou — APPROVED** [review]
LGTM, but I'd wait for @kou 's approval

**Review by kou — APPROVED** [review]
+1

**Comment by amoeba** [comment]
Thanks for the reminder @pitrou and for the reviews, both you and @kou.

**Pull request merged**
